modules/social_features/social_user/src/Plugin/Action/SocialSendEmail.php

```diff
@@ -0,0 +1,198 @@
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Utility\Token $token
+   *   The token service.
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
+   *   The entity manager.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   A logger instance.
+   * @param \Drupal\Core\Mail\MailManagerInterface $mail_manager
+   *   The mail manager.
+   * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
+   *   The language manager.
+   * @param \Egulias\EmailValidator\EmailValidator $email_validator
+   *   The email validator.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, Token $token, EntityManagerInterface $entity_manager, LoggerInterface $logger, MailManagerInterface $mail_manager, LanguageManagerInterface $language_manager, EmailValidator $email_validator) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->token = $token;
+    $this->storage = $entity_manager->getStorage('user');
+    $this->logger = $logger;
+    $this->mailManager = $mail_manager;
+    $this->languageManager = $language_manager;
+    $this->emailValidator = $email_validator;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static($configuration, $plugin_id, $plugin_definition,
+      $container->get('token'),
+      $container->get('entity.manager'),
+      $container->get('logger.factory')->get('action'),
+      $container->get('plugin.manager.mail'),
+      $container->get('language_manager'),
+      $container->get('email.validator')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute($entity = NULL) {
+    /** @var \Drupal\Core\Entity\Entity $entity */
+    if (!$entity->getEntityTypeId() === 'user') {
+      $this->logger->notice('Can not send e-mail for %entity', ['%entity' => $entity->getEntityTypeId() . ':' . $entity->id()]);
+      return;
+    }
+    /** @var \Drupal\user\Entity\User $entity */
+    if ($entity) {
+      $langcode = $entity->getPreferredLangcode();
+    }
+    else {
+      $langcode = $this->languageManager->getDefaultLanguage()->getId();
+    }
+    $params = ['context' => $this->configuration];
+
+    $message = $this->mailManager->mail('system', 'action_send_email', $entity->getEmail(), $langcode, $params);
+    // Error logging is handled by \Drupal\Core\Mail\MailManager::mail().
+    if ($message['result']) {
+      $this->logger->notice('Sent email to %recipient', ['%recipient' => $entity->getEmail()]);
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildPreConfigurationForm(array $form, array $values, FormStateInterface $form_state) {
```

Comment:
@jaapjan delete empty function.

Reply:
If i do that then i get this warning:
`Class must be declared abstract or implement method 'buildPreConfigurationForm'` so I just checked at what VBO example did and implemented the same logic.